internal/pkg/vm/internal/vmcontext/syscalls.go

```diff
@@ -17,8 +17,8 @@ type syscalls struct {
 var _ specsruntime.Syscalls = (*syscalls)(nil)
 
 // VerifySignature implements Syscalls.
-func (sys syscalls) VerifySignature(signature specscrypto.Signature, signer address.Address, plaintext []byte) bool {
-	return crypto.IsValidSignature(plaintext, signer, signature)
+func (sys syscalls) VerifySignature(signature specscrypto.Signature, signer address.Address, plaintext []byte) error {
```

Comment:
All changes to signature code stem from here. The syscalls interfaces expects VerifySignature to return an error. I performed the change here and bubbled it up through the rest of the code.